pilot/pkg/serviceregistry/serviceentry/conversion.go

```diff
@@ -140,6 +141,15 @@ func ServiceToServiceEntry(svc *model.Service, proxy *model.Proxy) *config.Confi
 // convertServices transforms a ServiceEntry config to a list of internal Service objects.
 func convertServices(cfg config.Config, clusterID cluster.ID) []*model.Service {
 	serviceEntry := cfg.Spec.(*networking.ServiceEntry)
+	// ShouldV2AutoAllocateIP already checks that there are no addresses in the spec however this is critical enough to likely be worth checking
```

Comment:
Hmmm. My initial thought was that we should have a smaller set of checks here, such that they could set the label to opt-out of _Istio_ assigning the address, and assign it through some other mechanism. But then if they do that, it may actually be valid with NONE mode.

So I wonder if we need any check at all (beyond `len(spec.Address) == 0`)?

Reply:
That does assume we remove the address if they opt out/make it NONE

Reply:
Though changing to NONE would have a race condition there